### PR TITLE
Adds username auth for Redis 6

### PR DESCRIPTION
**Pull request by scuml**
Adds optional username AUTH argument to connect to Redis 6 instances using ACL

### CODE REVIEW

File: Sources/UniRedis/UniRedis.swift [hunk 1]
```diff
@@ -53,27 +54,31 @@ public class UniRedis {
 	}
 
 	public init(_ url: String) throws {
-		guard let match = try url.match("^(?:redis(\\+sentinel)?://)?(?:([^@]+)@)?([^:/]+)(?::([0-9]+))?(?:/([0-9]+))?$"), match.count == 5 else {
+		guard let match = try url.match("^(?:redis(\\+sentinel)?://)?(?:([^:@]+):)?(?:([^@]+)@)?([^:/]+)(?::([0-9]+))?(?:/([0-9]+))?$"), match.count == 6 else {
```
Comment (danielfojt) [inline Sources/UniRedis/UniRedis.swift]: This seems to break when redis port number is present, ex. redis://user:pass@127.0.0.1:6380/
Let's keep original regexp and split user:pass later, in the code below, see next comment. This way, we could also avoid renumbering match[] indexes.

File: Sources/UniRedis/UniRedis.swift [hunk 2]
```diff
@@ -53,27 +54,31 @@ public class UniRedis {
...
+		if let m = match[5] {
 			db = Int(m)!
 		}
-		if let m = match[1] {
+		if let m = match[2] {
 			password = m
 		}
+		if let m = match[1] {
+			username = m
+		}
```
Comment (danielfojt) [inline Sources/UniRedis/UniRedis.swift:75]: As commented above, let's handle user:pass as a single token and try to split it here, like:

```swift
if let m = match[1] {
     if let auth = try m.match("^(.+):(.+)$") {
          username = auth[1]
          password = auth[2]
     } else {
          password = m
     }
}
```

Just proposal, not tested.

**Comment by scuml** [comment]
Sounds good - this is ready to go.

**Comment by danielfojt** [comment]
Nice, thank you very much, Stephen.

**Pull request merged**